packages/federation/src/supergraph.ts

```diff
@@ -1676,11 +1680,26 @@ function mergeResults(results: unknown[], getFieldNames: () => Set<string>) {
     if (datas.length === 1) {
       return makeExternalObject(datas[0], errors, getFieldNames);
     }
-    return makeExternalObject(
-      mergeDeep(datas, undefined, true, true),
-      errors,
-      getFieldNames,
-    );
+    const mergedData = mergeDeep(datas, undefined, true, true);
+    // Put original symbols on the merged object
+    const symbols = [
+      OBJECT_SUBSCHEMA_SYMBOL,
+      FIELD_SUBSCHEMA_MAP_SYMBOL,
+      UNPATHED_ERRORS_SYMBOL,
+    ];
+    for (const symbol of symbols) {
+      if (mergedData?.[symbol] == null) {
+        for (const data of datas) {
+          // @ts-expect-error - we know it is there
+          const symbolValue = data?.[symbol];
+          if (symbolValue != null) {
+            mergedData[symbol] = symbolValue;
+            break;
+          }
+        }
+      }
+    }
```

Comment:
To avoid this, in the `withState` util, I'm copying property descriptors instead of attributes.

This way, you can copy everything, including properties that have a getter instead of an actual value